pkg/cache/metrics.go

```diff
@@ -0,0 +1,56 @@
+package cache
+
+import (
+	"context"
+	"time"
+
+	"github.com/rancher/lasso/pkg/metrics"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+const (
+	cacheMetricsCollectionPeriod = 1 * time.Minute
+)
+
+func (f *sharedCacheFactory) collectMetrics() sharedCacheFactoryMetrics {
+	f.lock.RLock()
+	defer f.lock.RUnlock()
```

Comment:
I meant to add, I suspect this should be a `.Lock()` not a `.RLock()`

https://go.dev/ref/spec#For_range

> If a map entry is created during iteration, that entry may be produced during the iteration or may be skipped.

Reply:
The lock protects `f.caches`, not the map being created. Given it was already protected, I only changed it to a `RWMutex`. It's unlikely to happen, but this may panic if a new cache is added/started so `f.caches` gets a write while reading here